prometheus/exporter/docker_stats.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python
+import subprocess
+import json
+import sys
+import re
+
+def parsePercentile(data):
+    return data.replace("%", "")
+
+def parseIO(data):
+    inOut = data.split("/")
+    inByte = convertToByte(inOut[0])
+    outByte = convertToByte(inOut[1])
+    return {"in": inByte, "out": outByte}
+
+def parseUsageLimit(data):
+    usageLimit = data.split("/")
+    usageByte = convertToByte(usageLimit[0])
+    limitByte = convertToByte(usageLimit[1])
+    return {"usage": usageByte, "limit": limitByte}
+
+def convertToByte(data):
+    data = data.lower()
+    number = float(re.findall(r"\d", data)[0])
+    if ("tb" in data) or ("tib" in data):
+        return number * 1024 * 1024 * 1024 * 1024
+    elif ("gb" in data) or ("gib" in data):
+        return number * 1024 * 1024 * 1024
+    elif ("mb" in data) or ("mib" in data):
+        return number * 1024 * 1024
+    elif ("kb" in data) or ("kib" in data):
+        return number * 1024
+    else: 
+        return number
+
+def parseDockerStats(stats):
+    data = [line.split(',') for line in stats.splitlines()]
+    # pop the headers
+    data.pop(0)
+    rowNum = len(data)
+    colNum = len(data[0])
+    containerStats = {}
+
+    for i in range(rowNum):
+        id = data[i][1]
+        containerInfo = {
+            "Container": data[i][0],
```

Comment:
>Container [](start = 13, length = 9)

what does this mean? container name?

Reply:
dockerStatsCMD = "sudo docker stats --no-stream --format \"table {{.Container}}, {{.ID}}, {{.CPUPerc}},{{.MemUsage}},{{.NetIO}},{{.BlockIO}},{{.MemPerc}}\""

query results example:
CONTAINER,CONTAINER ID,CPU %,MEM USAGE / LIMIT,NET I/O,BLOCK I/O,MEM %
33a22dcd4ba3,33a22dcd4ba31ebc4a19fae865ee62285b6fae98a6ab72d2bc65e41cdc70e419,0.00%,18.68MiB / 300MiB,0B / 0B,28MB / 0B,6.23%

container is the short container id which will show at cmd results